submissions/SpidertronWeaponSwitcher/SpidertronWeaponSwitcher.md

```diff
@@ -0,0 +1,19 @@
+# Spidertron Weapon Switcher - Under the hood
+### Xorimuth
+
+https://mods.factorio.com/SpidertronWeaponSwitcher
+
+[[TODO: insert gif here of spidertron fighting biter bases whilst switching between different weapons]]
+
+If you’ve never written a mod before, you may not know that many things in Factorio cannot be changed _during the game_, for example, the range of a gun, the graphics of a building and, importantly for me, the weapon slots that a vehicle has. Of course, the weapons that a spidertron has _can_ be changed by mods, but only by creating a new ‘type’ of spidertron during the game’s loading phase.
+
+The way that I got around this limitation is by creating 5 ‘types’ of spidertron, each one with a different set of weapons. When you switch the weapons of a spidertron, what the mod actually does is:
+
+1) Store all the data about the spidertron (such as its inventory contents and position)
+2) Delete the spidertron
+3) Create a new spidertron of the next ‘type’ (with different weapon slots)
+4) Insert all the stored data into the new spidertron
```

Comment:
Insert snippet of code. Factorio readers would love that

Reply:
I can't really find any one section of code that would be relevant. The process of saving and retrieving the data is not very pretty imo. Perhaps [lines 100-102](https://github.com/tburrows13/SpidertronWeaponSwitcher/blob/v1.0.0/control.lua#L100-L102) or [56-62](https://github.com/tburrows13/SpidertronWeaponSwitcher/blob/v1.0.0/control.lua#L56-L64) (without the comment)?

In case anyone's interested in more technical details:
https://github.com/tburrows13/SpidertronWeaponSwitcher/blob/v1.0.0/control.lua
- `store_spidertron_data()` returns a table of all the data
- then `replace_spidertron()` does the deleting and placing of the new entity (and also handles the saving and retrieving of the ammo from `global` - not discussed in this post)
- finally, `place_stored_spidertron_data()` uses the table from `store_spidertron_data()` to repopulate the spidertron

Reply:
I can't really find any one section of code that would be relevant. The process of saving and retrieving the data is not very pretty imo. Perhaps [lines 100-102](https://github.com/tburrows13/SpidertronWeaponSwitcher/blob/v1.0.0/control.lua#L100-L102) or [56-62](https://github.com/tburrows13/SpidertronWeaponSwitcher/blob/v1.0.0/control.lua#L56-L64) (without the comment)?

In case anyone's interested in more technical details:
https://github.com/tburrows13/SpidertronWeaponSwitcher/blob/v1.0.0/control.lua
- `store_spidertron_data()` returns a table of all the data
- then `replace_spidertron()` does the deleting and placing of the new entity (and also handles the saving and retrieving of the ammo from `global` - not discussed in this post)
- finally, `place_stored_spidertron_data()` uses the table from `store_spidertron_data()` to repopulate the spidertron